### PR TITLE
feat: add support for Decimal scalar

### LINKED ISSUE #94: Add support for Decimal scalar
#### Perceived Problem

- https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#decimal
- Not supported in our custom scalar support

#### Ideas / Proposed Solution(s)

- https://stackoverflow.com/questions/35709595/why-would-you-use-a-string-in-json-to-represent-a-decimal-number (via @iddan)

### PR DESCRIPTION
**Pull request by iddan**
closes #94 
This PR includes copied code from https://github.com/unlight/prisma-graphql-type-decimal with attribution

#### TODO

- [x] docs
- [x] tests

### CODE REVIEW

**Comment by jasonkuhrt** [comment]
@iddan will need some rebasing

**Comment by lvauvillier** [comment]
Hi @iddan, @jasonkuhrt. Prisma client already use decimal.js internally.
We dont need to add it as dependency and use the same version as Prisma.

```typescript
import { Prisma } from "@prisma/client";

Prisma.Decimal(value);
```

**Comment by iddan** [comment]
The problem is Prisma is not available when using @prisma/client as a lib dependency

**Comment by jasonkuhrt** [comment]
@iddan not sure what you mean?

**Comment by iddan** [comment]
When I tried to use `Prisma.Decimal` the tests failed as `@prisma/client` doesn’t have  a `Prisma` member until you run `prisma generate`

**Comment by jasonkuhrt** [comment]
@iddan Ah I see. That technically should be acceptable since NP is not usable until PC has been generated anyways. Still, let's pin to our own version of decimal.js because its the correct thing to do--deps should not be implicit generally speaking.

Also we [will] test and support a set of Prisma versions for this kind reason. We should/will know from tests if the decimal JS we use and the one used by PC is not compatible ever one day.

**Comment by iddan** [comment]
Is there an action item for me for this PR to get merged?

File: src/entrypoints/scalars.ts [hunk 1]
```diff
@@ -13,6 +14,7 @@ import { Json } from '../scalars/Json'
  * | `DateTime` | `DateTime` | `dateTime`       | [DateTime](https://www.graphql-scalars.dev/docs/scalars/datetime)     |                                                                                                              |
  * | `BigInt`   | `BigInt`   | `bigInt`         | [BigInt](https://www.graphql-scalars.dev/docs/scalars/big-int)        | [JavaScript BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) |
  * | `Bytes`    | `Bytes`    | `bytes`          | [Byte](https://www.graphql-scalars.dev/docs/scalars/byte/)            | [Node.js Buffer](https://nodejs.org/api/buffer.html#buffer_buffer)                                           |
+ * | `Decimal` | `Decimal` | `decimal` | | |.
```
Comment (jasonkuhrt) [inline src/entrypoints/scalars.ts:17]:
```suggestion
 * | `Decimal` | `Decimal` | `decimal` | (internal) | Uses [Decimal.js](https://mikemcl.github.io/decimal.js) |.
```

File: tests/e2e/e2e.test.ts [hunk 2]
```diff
@@ -155,6 +156,7 @@ it('When bundled custom scalars are used the project type checks and generates e
             data: {
               id: 'foo1',
               someDateTimeField: new Date("2021-05-10T20:42:46.609Z"),
+              someDecimalField: new Prisma.Decimal(24.454545),
```
Comment (jasonkuhrt) [inline tests/e2e/e2e.test.ts:159]: Create this decimal value using the `decimal.js` dependency instead of Prisma's bundled version.

**Review by jasonkuhrt — CHANGES_REQUESTED** [review]
Almost there!

**Comment by iddan** [comment]
I will issue the comments ASAP

File: src/scalars/Decimal.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    serialize(value: decimal.Decimal) {
+      return value.toString()
+    },
+    /**
+     * Value from the client
+     */
+    parseValue(value: decimal.Decimal.Value) {
+      return new decimal.Decimal(value)
+    },
+    parseLiteral(ast) {
```
[the remark this answers is not in the document]
Comment (iddan) [reply src/scalars/Decimal.ts:54]: I just copied it from [prisma-graphql-type-decimal](https://www.npmjs.com/package/prisma-graphql-type-decimal)

**Comment by jasonkuhrt** [comment]
@iddan do you know what's up with the newly failing tests?

File: src/scalars/Decimal.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jasonkuhrt) [reply src/scalars/Decimal.ts:54]: Yeah figured, I guess I can look up up the GraphQL package API to learn more :) just being lazy.

**Comment by iddan** [comment]
No idea, maybe it has to do with the new Prisma release?

**Comment by jasonkuhrt** [comment]
@iddan but you didn't touch the Prisma version in this PR did you?

CC @Jolg42 @janpio if you know of any recent windows related issues or changes let us know 🙏

**Comment by iddan** [comment]
I haven’t

**Comment by janpio** [comment]
I am not aware of anything. Possible that Prisma changed some path handling, but that should be visible when you test your existing codebase with that version as well and not just in this PR.

(I have no indication or reproduction of the new Prisma breaking any old things, as the old packages are completely untouched and the only external dependency, the binaries, do not influence any of this - and also did not change of course)

**Review by jasonkuhrt — APPROVED** [review]
Fixed the test problem on trunk, this pr LGTM!

**Comment by jasonkuhrt** [comment]
I will fix the snapshot failures tomorrow

**Pull request merged**